File: backend/src/main/java/com/paperai/model/entity/Paper.java
package com.paperai.model.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 论文实体
 *
 * @author: ch
 * @date 2026年05月11日
 */
@Data
@TableName("paper")
public class Paper implements java.io.Serializable {
    private static final long serialVersionUID = 1L;

    /** 论文ID */
    @TableId(type = IdType.AUTO)
    private Long id;

    /** 论文标题 */
    private String title;

    /** 摘要 */
    private String abstractText;

    /** 关键词 */
    private String keywords;

    /** 研究方向/描述 */
    private String description;

    /** 论文状态（DRAFT/REVIEWING/PUBLISHED） */
    private String status;

    /** 最终内容（仅运行时从版本表填充，不持久化到 paper 表） */
    @TableField(exist = false)
    private String content;

    private Long userId;

    /** 当前最新版本号 */
    private Integer currentVersion;

    /** 创建时间 */
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    /** 更新时间 */
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
}

File: backend-py/app/models/paper.py
from sqlalchemy import String, Text, Integer, BigInteger
from sqlalchemy.orm import Mapped, mapped_column
from .base import Base, TimestampMixin


from sqlalchemy import Index

class Paper(Base, TimestampMixin):
    __tablename__ = "paper"
    __table_args__ = (
        Index("ix_paper_user_id", "user_id"),
        Index("ix_paper_status", "status"),
    )

    id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
    title: Mapped[str] = mapped_column(String(300), nullable=False)
    abstract_text: Mapped[str | None] = mapped_column(Text)
    keywords: Mapped[str | None] = mapped_column(String(500))
    description: Mapped[str | None] = mapped_column(Text)
    status: Mapped[str] = mapped_column(String(20), nullable=False, default="DRAFT")
    user_id: Mapped[int] = mapped_column(BigInteger, nullable=False, default=0)
    current_version: Mapped[int] = mapped_column(Integer, nullable=False, default=0)
    kg_id: Mapped[int | None] = mapped_column(BigInteger)
